src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -42,6 +42,12 @@ enum TrimDirection {
     Trailing,
 }
 
+enum TargetLen {
+    Invalid,
+    Zero,
```

Comment:
Why not `Len(0)`?

Reply:
So this can be replaced with `Option<usize>`.

Reply:
actually ,there is another value , when max_allowed_packet is pushed down, if target_len is greater than max_allowed_packet,  it should tell the pad function to return null with an warning.
but for now, Option is enough .